include/swift/AST/DiagnosticsSema.def

```diff
@@ -2411,6 +2411,14 @@ NOTE(force_optional_to_any,none,
      "force-unwrap the value to avoid this warning", ())
 NOTE(silence_optional_to_any,none,
      "explicitly cast to Any with 'as Any' to silence this warning", ())
+WARNING(optional_in_string_interpolation_segment,none,
+        "optional value in string interpolation can have unintended behavior",
+        ())
```

Comment:
"unintended behavior" still seems to vague to me.

Can you try to find something more specific to use here?